services/cloud/availabilityzone/availabilityzone.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache v2.0 License.
+
+package availabilityzone
+
+import (
+	"github.com/microsoft/moc-sdk-for-go/services/cloud"
+	"github.com/microsoft/moc/pkg/errors"
+	"github.com/microsoft/moc/pkg/status"
+	wssdcloudcompute "github.com/microsoft/moc/rpc/cloudagent/cloud"
+)
+
+// this is only used in create/update
+func getRpcAvailabilityZone(s *cloud.AvailabilityZone) (*wssdcloudcompute.AvailabilityZone, error) {
+	if s == nil {
+		return nil, errors.Wrapf(errors.InvalidInput, "Availability zone object is nil")
+	}
+
+	if s.Name == nil {
+		return nil, errors.Wrapf(errors.InvalidInput, "AvailabilityZone object Name is empty")
+	}
+
+	availabilityZone := &wssdcloudcompute.AvailabilityZone{
+		Name:                     *s.Name,
+		Status:                   status.GetFromStatuses(s.Statuses),
```

Comment:
In Create / Update flow , is the 'Status' field required to be set ? Since the request is going from moccli to cloud agent

Reply:
no status field is not a required parameter from moccli